onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp

```diff
@@ -56,11 +59,16 @@ OMStatus OMTrainingHandler::handleError(const OMConfig &config, OMRuntimeStorage
     OMStatus status = forward_storage.getDataByTensorIndex(&calculated_data, forward_output_index);
     assert(calculated_data != nullptr);
 
+    OMLoss loss_type = config.training_context.loss;
+
     // Get target data
     auto data_type_size = sizeof(core::OMDataType(forward_output_tensor->type()));
     size_t offset = batch_num * data_type_size * flat_size;
+    if (loss_type == SPARSE_CROSS_ENTROPY)
+    {
+        offset = batch_num * data_type_size;
+    }
```

Comment:
@BalyshevArtem 
Is it ok to changing offset according to the loss_type??
I had no idea of refactoring this code to apply sparse cross entropy... :'(